Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -1195,6 +1199,14 @@ void GetHitTestResultsLegacy(std::vector<HitResult>& filteredResults, xr::HitTes
         return m_impl->sessionImpl.GetPlaneByID(planeID);
     }
 
+    bool System::Session::Frame::IsTracking() const {
+        // There are three different tracking states as defined in ARKit: https://developer.apple.com/documentation/arkit/artrackingstate
+        // From my testing even while obscuring the camera for a long duration the state still registers as ARTrackingStateLimited
+        // rather than ARTrackingStateNotAvailable. For that reason the only state that should be considered to be trully tracking is
+        // ARTrackingStateNormal.
+        return m_impl->sessionImpl.GetTrackingState() == ARTrackingState::ARTrackingStateNormal;
```

Comment:
I would do this inside the impl and call it `IsTracking`. The purpose of the impl is just to hide the private members so that different implementations can have different implementation details. But for the most part, the wrapper should just be a wrapper and not introduce any additional logic when possible.

Reply:
Updated to move the logic into SessionImpl